pkg/relayer/proxy/server_builder.go

```diff
@@ -0,0 +1,57 @@
+package proxy
+
+import (
+	"context"
+
+	sharedtypes "pocket/x/shared/types"
+	suppliertypes "pocket/x/supplier/types"
+)
+
+type RelayServersMap map[string][]RelayServer
+
+// BuildProvidedServices builds the provided services from the supplier's on-chain advertised services.
+// It populates the relayerProxy's `providedServices` map of servers for each service, where each server
+// is responsible for listening for incoming relay requests and poxying them to the supported native service.
```

Comment:
Similar concerns about consistent naming. We have "advertised services", "providedServices", and "server" referring to the same object(s), as well as "proxying" in that same context; whereas, it seems like it should be "relaying" to avoid ambiguity between `RelayerProxy` and `RelayServer`.